src/charm.py

```diff
@@ -74,13 +74,12 @@ def _on_admin_relation_changed(self, event):
     @log_event_handler
     def _on_tctl_action(self, event):
         """Run the tctl command line tool."""
-        # TODO(frankban): make this work.
         container = self.unit.get_container(self.name)
         if not container.can_connect():
             event.fail("cannot connect to container")
             return
 
-        args = event.params["args"].split()
+        args = ["--address", "temporal-k8s:7233", *event.params["args"].split()]
```

Comment:
Is this using the internal k8s dns? If so, this would not work if you deploy temporal with an app name different than the default.

Reply:
Yes. I thought we would keep the defaults for now until we start to making the charm more configurable at a later stage.